Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-constraints/ProblemConstraints.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useCallback, useMemo } from 'react';
+import isNil from 'lodash/isNil';
+
+import { useProblems } from '../../../hooks/use-problems';
+import { format } from '../../../utils/number-utils';
+import IconSize from '../../guidelines/icons/common/icon-sizes';
+import QuestionIcon from '../../guidelines/icons/QuestionIcon';
+import List from '../../guidelines/lists/List';
+
+import styles from './ProblemConstraints.module.scss';
+
+const ProblemConstraints = () => {
+    const { state: { currentProblem } } = useProblems();
+
+    const constraints = useMemo(() => {
+        if (isNil(currentProblem)) {
+            return [];
+        }
+
+        const { memoryLimit, fileSizeLimit, timeLimit, checkerName } = currentProblem;
+
+        return [ `Allowed working time: ${format(timeLimit).toString()} sec.`,
+            `Allowed memory: ${format(memoryLimit).toString()} MB`,
+            `Size limit: ${format(fileSizeLimit).toString()} MB`,
+            `Checker: ${checkerName}` ];
+    }, [ currentProblem ]);
+
+    const renderConstraint = useCallback(
+        (constraint: string) => <span>{constraint}</span>,
+        [],
+    );
+
+    const getCheckerDescription = useMemo(() => {
```

Comment:
`checkerDescription`, without `get`

Reply:
Does it have to be in `useMemo`, why not just `const { checkerDescription = '' } = currentProblem || {};` Once the `currentProblem` is retrieved, the value will be "momoized" anyways, as we just get it from the object 🤔

Reply:
Better to use `useMemo` for all variables. This way if `currentProblem` is not `useMemo`ed, `checkerDescription` will still be memoized.